### DISCUSSION TITLE
Is the CDM Versioning working?

### DISCUSSION THREAD

**Discussion by chrisisla** [finos/common-domain-model#2305] [General]
A discussion on CDM versioning was undertaken in the TAWG meeting here: #2304 

The conclusion was that there was still some confusion/disagreement about the semantic version structure that was used, especially in relation to the development releases. The consensus was that the "major.minor.patch" structure worked. The "dev.nnn" on the end for development releases was questioned by some participants as it seemed superfluous to them.

This powerpoint on [Semantic Versioning.pdf](https://github.com/finos/common-domain-model/files/12315474/Semantic.Versioning.pdf) shows how the versioning works now and how to get to the correct versions on github. 

Please add any feedback to this discussion. If nothing is raised by the next TAWG meeting in September then it will be assumed that the current version structure is fine and this discussion will be closed.

**Comment by dshoneisda** [comment]
Chris- if there are no issues raised and nothing changes, is the the intention to make that pdf available and prominent on the repo for people when they first arrive? (assuming it isn't already)

**Comment by lolabeis** [comment]
There is already a section in the CDM doc that presents the sem-ver approach:
https://cdm.finos.org/docs/contribution/#semantic-versioning
I suggest that the pdf be integrated / merged with it, to avoid any inconsistency / duplicate.

If we want a versioning explanation document to feature prominently in the repository, that document could be linked / referenced in the CDM doc. For instance, the diagram in the pdf is quite a handy summary. It could be saved as a stand-alone image in the repo and a link to that image could be inserted in the main doc.

**Comment by dschwartznyc** [comment]
While the clarification of CDM's use of semantic version is very helpful, there are other considerations about whether the approach is working.  Note that some of these may overlap with questions about the build process.

- When and how are new versions published?
- Is there a new version for every model change?
- If a collection of changes is bundled into a version, how can one find a summary of all the changes?
- How are changes allocated to a version?  
- Are all non-breaking changes included in the current major version?
- Is there a mechanism to ensure that changes targeted at the current major version are non-breaking?
- For each version, where does one find all the relevant assets related to that release (documentation, source Rosetta, generated code, sample JSON files compatible with the version, et al)?
- Is the use of "-dev" in the numbering scheme consistent with best practices?  Note that it may be a breaking protocol for some of the implementation languages (I believe that Python expects a version label in the format #.#.#).
- What is the process for ending support for a version?  Who decides?  How is the decision communicated?

**Comment by lolabeis** [comment]
Closing this discussion, which appears superseded in all aspects by some of the components being discussed in #2774.
